iterators and generators 검토 (#190)
yeonjuan/Typescript-Handbook-ko: iterators-and-generators.md

# 이터러블 (Iterables)

객체가 [`Symbol.iterator`](https://www.typescriptlang.org/docs/handbook/symbols.html#symboliterator)프로퍼티에 대한 구현을 가지고 있다면 이터러블로 간주합니다.
`Array`, `Map`, `Set`, `String`, `Int32Array`, `Uint32Array` 등과 같은 일부 내장 타입에는 이미 `Symbol.iterator` 프로퍼티가 구현되어 있습니다.
객체의 `Symbol.iterator` 함수는 반복할 값 목록을 반환합니다.

## `for..of` 문

> `for..of`는 이터러블 객체를 반복하여, 객체의 `Symbol.iterator` 프로퍼티를 호출합니다.

Reviewer: [제안]
```suggestion
`for..of`는 객체의 `Symbol.iterator` 프로퍼티를 호출하여, 이터러블 객체를 반복합니다.
```

다음은 배열의 간단한 `for..of` 루프입니다:

```ts
let someArray = [1, "string", false];

for (let entry of someArray)
 console.log(entry); // 1, "string", false
}
```

### `for..of` vs. `for..in` 문

> `for..of` 및 `for..in` 문 모두 목록을 반복합니다; 반복되는 값은 다르지만, `for..in`은 반복되는 객체의 *키* 목록을 반환하고, 반면에 `for..of`는 반복되는 객체의 숫자 속성 *값* 목록을 반환합니다.

Reviewer 2: ```suggestion
`for..of` 및 `for..in` 문 모두 목록을 반복합니다; 반복되는 값은 다르지만, `for..in`은 반복되는 객체의 *키* 목록을 반환하고, 반면에 `for..of`는 반복되는 객체의 숫자 프로퍼티 *값* 목록을 반환합니다.
```

}
```

또 다른 차이점은 `for..in`은 모든 객체에서 작동한다는 것입니다; 객체의 프로퍼티를 검사하는 방법으로 사용됩니다.

> 반면에 `for..of`는 이터러블 객체의 값에 주로 관심이 있습니다. `Map` 및 `Set`과 같은 내장 객체는 저장된 값에 접근할 수 있는 `Symbol.iterator` 프로퍼티를 구현합니다.

Reviewer 3: ```suggestion
반면에 `for..of`는 주로 관심이 이터러블 객체의 값에 있습니다. `Map` 및 `Set`과 같은 내장 객체는 저장된 값에 접근할 수 있는 `Symbol.iterator` 프로퍼티를 구현합니다.
```
